src/api/api-types.ts

```diff
@@ -0,0 +1,324 @@
+  temperature?: number;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "RewriteResponse".
+ */
+export interface RewriteResponse {
+  status: ResponseStatus;
+  message: string;
+  choices: RewriteChoice[];
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "S2SearchResult".
+ */
+export interface S2SearchResult {
+  title?: string;
+  abstract?: string;
+  venue?: string;
+  year?: number;
+  paperId?: string;
+  citationCount?: number;
+  openAccessPdf?: string;
+  authors?: string[];
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "SearchRequest".
+ */
+export interface SearchRequest {
+  query: string;
+  limit?: number;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "SearchResponse".
+ */
+export interface SearchResponse {
+  status: ResponseStatus;
+  message: string;
+  results: S2SearchResult[];
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "TextCompletionChoice".
+ */
+export interface TextCompletionChoice {
+  index: number;
+  text: string;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "TextCompletionRequest".
+ */
+export interface TextCompletionRequest {
+  text: string;
+  n_choices?: number;
+  temperature?: number;
+  max_tokens?: number;
+  title?: string;
+  abstract?: string;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "TextCompletionResponse".
+ */
+export interface TextCompletionResponse {
+  status: ResponseStatus;
+  message: string;
+  choices: TextCompletionChoice[];
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "UpdateStatusResponse".
+ */
+export interface UpdateStatusResponse {
+  status: ResponseStatus;
+  message: string;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "Body_create_file__project_id___filepath__put".
+ */
+export interface BodyCreateFile_ProjectId__Filepath_Put {
+  file: File;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "LoggingSettings".
+ */
+export interface LoggingSettings {
+  enable?: boolean;
+  filepath?: string;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "OpenAISettings".
+ */
+export interface OpenAISettings {
+  api_key?: string;
+  chat_model?: string;
+  manner?: RewriteMannerType & string;
+  temperature?: number;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "ProjectSettings".
+ */
+export interface ProjectSettings {
+  current_directory?: string;
+}
+/**
+ * This interface was referenced by `ApiSchema`'s JSON-Schema
+ * via the `definition` "SettingsSchema".
+ */
+export interface SettingsSchema {
```

Comment:
These are all the types that appear in a FastAPI request/response type or are referenced from one of those types. Note how all the fields are optional, which is not what we want.